core/server/master/src/test/java/alluxio/master/meta/UpdateCheckTest.java

```diff
@@ -46,86 +53,135 @@ public void before() {
     Mockito.when(EnvironmentUtils.isGoogleComputeEngine()).thenReturn(true);
     Mockito.when(EC2MetadataUtils.getUserData())
         .thenThrow(new SdkClientException("Unable to contact EC2 metadata service."));
 
-    String userAgentString = UpdateCheck.getUserAgentString("cluster1");
-    Assert.assertTrue(userAgentString.equals(
-        String.format("Alluxio/%s (cluster1; gce)", ProjectConstants.VERSION)));
+    String userAgentString = UpdateCheck.getUserAgentEnvironmentString("cluster1");
+    Assert.assertTrue(userAgentString.equals("cluster1; gce"));
   }
 
   @Test
-  public void userAgentStringEC2AMI() throws Exception {
+  public void userAgentEnvironmentStringEC2AMI() throws Exception {
     Mockito.when(EnvironmentUtils.isEC2()).thenReturn(true);
     Mockito.when(EnvironmentUtils.getEC2ProductCode()).thenReturn("random123code");
     // When no user data in this ec2, null is returned
     Mockito.when(EC2MetadataUtils.getUserData()).thenReturn(null);
 
-    String userAgentString = UpdateCheck.getUserAgentString("cluster1");
-    Assert.assertTrue(userAgentString.equals(
-        String.format("Alluxio/%s (cluster1; ProductCode:random123code; ec2)",
-            ProjectConstants.VERSION)));
+    String userAgentString = UpdateCheck.getUserAgentEnvironmentString("cluster1");
+    Assert.assertTrue(userAgentString.equals("cluster1; ProductCode:random123code; ec2"));
   }
 
   @Test
-  public void userAgentStringEC2CFT() throws Exception {
+  public void userAgentEnvironmentStringEC2CFT() throws Exception {
     Mockito.when(EnvironmentUtils.isEC2()).thenReturn(true);
     Mockito.when(EnvironmentUtils.getEC2ProductCode()).thenReturn("random123code");
     Mockito.when(EnvironmentUtils.isCFT(Mockito.anyString())).thenReturn(true);
     Mockito.when(EC2MetadataUtils.getUserData()).thenReturn("{ \"cft_configure\": {}}");
 
-    String userAgentString = UpdateCheck.getUserAgentString("cluster1");
-    Assert.assertTrue(userAgentString.equals(
-        String.format("Alluxio/%s (cluster1; ProductCode:random123code; cft; ec2)",
-            ProjectConstants.VERSION)));
+    String userAgentString = UpdateCheck.getUserAgentEnvironmentString("cluster1");
+    Assert.assertTrue(userAgentString.equals("cluster1; ProductCode:random123code; cft; ec2"));
   }
 
   @Test
-  public void userAgentStringEC2EMR() throws Exception {
+  public void userAgentEnvironmentStringEC2EMR() throws Exception {
     Mockito.when(EnvironmentUtils.isEC2()).thenReturn(true);
     Mockito.when(EnvironmentUtils.getEC2ProductCode()).thenReturn("random123code");
     Mockito.when(EnvironmentUtils.isEMR(Mockito.anyString())).thenReturn(true);
     Mockito.when(EC2MetadataUtils.getUserData()).thenReturn("emr_apps");
 
+    String userAgentString = UpdateCheck.getUserAgentEnvironmentString("cluster1");
+    Assert.assertTrue(userAgentString.equals("cluster1; ProductCode:random123code; emr; ec2"));
+  }
+
+  @Test
+  public void featureString() {
```

Comment:
should this be split up into different unit tests for each individual feature to be consistent with the existing unit tests?

Reply:
addressed